tests/yast2_gui/yast2_lan_restart_vlan.pm

```diff
@@ -49,7 +49,7 @@ sub run {
     $network_settings->save_changes();
     wait_for_xterm_to_be_visible();
     clear_journal_log();
-    check_network_status('', 'vlan');
+    check_network_status('no_restart', 'vlan');
```

Comment:
when you pass `''` I thought that it should be `undef` inside the method and `//=` [logical-defined-or operator](https://perldoc.perl.org/perlop.html#Logical-Defined-Or) should do its work, but if it is not `undef` any other thing like empty string will not work, so could you remove the line with the `//=` as it doesn't make sense.

Reply:
The line works if check_network_status() is called without any arguments. I am thinking about changing the function parameters from 
my ($expected_status, $device) = @_;
to 
my $args = shift ;
and then we can define check_network_status({device=>'bridge'}); for example. YDYT:

Reply:
I vote for just remove the line.

Reply:
Alrighty then, I removed the line. I checked if function is used elsewhere, but all calls were already changed in previous commit to "check_network_status('no_restart',...."

Reply:
It seems that there is an issue when I remove that line due to line 201 in lib/y2lan_restart_common.pm. 
It is called by 'verify_network_configuration' which is used in 'tests/x11/yast2_lan_restart.pm' with no parameters given. So, maybe it's better to leave the line as it was.

Reply:
So you have a subroutine which expect a parameter and is designed in such a way that assume that sometimes it will be undefined (and then for those cases it gets a default one). Logically, it should be used as simple as taking only the $device. But this is not happening. Now we can say that the subroutine will always be used with both parameters. But this produces inconsistency and complicates things. What i am suggesting is to switch the parameters such as the $device is always passed and if the $expected_status is not used, it would take the default value. 
It is not unusual to have a change and realize that you have to  touch and fix other components too. So to make a change in  'tests/x11/yast2_lan_restart.pm' and wherever it needs to, should not be a big issue. We should try to make cleaner code and improve designs for the sake of the future usage. 
for me i cant approve but because i do not know what it is the approach and the policy of the team on the subject and obviously the current solution works and i know that you gave a lot of effort, i will leave it to the one who will merge it.

Reply:
> It seems that there is an issue when I remove that line due to line 201 in lib/y2lan_restart_common.pm.
> It is called by 'verify_network_configuration' which is used in 'tests/x11/yast2_lan_restart.pm' with no parameters given. So, maybe it's better to leave the line as it was.
you're right, it is really working, in all the calls to `verify_network_configuration` as `$expected_status` is not passed, so inside `check_network_status` get always undef which is translated fine to `no_restart`. Besides that is also called with `$workaround` which doesn't make sense or is a parameter that is not used anymore. So you fixed properly, just avoiding to call it with `''` it should be call with `undef` or explicitly like you did. The main problem is that those functions do not have named parameters and it is kind of playing golf passing variables, and here is not the point to fix that or delete that assignation that I mentioned which could break something else.

What we normally do is improve existing code if is quick and straightforward to do it and we don't really invest time in refactor those old UI test as they a bad design that has been patched many time (by me and by others), except for fixing it when failing (and if fails too much, then we redesign them), for that reason I mentioned that was 'nice to have'.